File: classifier/data/raw/ai/samples/project_euler_java/Problem064.java
// Project Euler 64: odd periods in continued fractions of square roots.

public class Problem064 {
    public static void main(String[] args) {
        int count = 0;
        for (int d = 2; d <= 10_000; d++) {
            int root = (int) Math.sqrt(d);
            if (root * root == d) continue;
            int m = 0, denominator = 1, a = root, period = 0;
            do {
                m = denominator * a - m;
                denominator = (d - m * m) / denominator;
                a = (root + m) / denominator;
                period++;
            } while (a != 2 * root);
            if ((period & 1) == 1) count++;
        }
        System.out.println(count);
    }
}

File: classifier/data/raw/human/samples/project_euler_zig/problem_064.zig
const std = @import("std");

// Project Euler 64: odd continued-fraction periods of square roots up to 10000.
fn isSquare(n: u32) bool {
    var root: u32 = 1;
    while (root * root < n) : (root += 1) {}
    return root * root == n;
}

pub fn main() !void {
    var odd_periods: u32 = 0;
    for (2..10_001) |n_usize| {
        const n: u32 = @intCast(n_usize);
        if (isSquare(n)) continue;
        var a0: u32 = 1;
        while ((a0 + 1) * (a0 + 1) <= n) : (a0 += 1) {}
        var m: u32 = 0;
        var denominator: u32 = 1;
        var a: u32 = a0;
        var period: u32 = 0;
        while (true) {
            m = denominator * a - m;
            denominator = (n - m * m) / denominator;
            a = (a0 + m) / denominator;
            period += 1;
            if (a == 2 * a0) break;
        }
        if (period % 2 == 1) odd_periods += 1;
    }
    var buffer: [32]u8 = undefined;
    const text = try std.fmt.bufPrint(&buffer, "{d}\n", .{odd_periods});
    _ = std.c.write(1, text.ptr, text.len);
}
